daemon/containerd/resolver.go

```diff
@@ -132,3 +134,27 @@ func (f httpFallback) RoundTrip(r *http.Request) (*http.Response, error) {
 
 	return resp, err
 }
+
+func isTLSError(err error) bool {
+	var tlsErr tls.RecordHeaderError
+	if errors.As(err, &tlsErr) && string(tlsErr.RecordHeader[:]) == "HTTP/" {
+		return true
```

Comment:
Perhaps keep the comment here (we could add the same to the containerd code), wdyt?;

```
// server gave HTTP response to HTTPS client
```